src/mon/OSDMonitor.cc

```diff
@@ -75,6 +75,59 @@
 #define dout_subsys ceph_subsys_mon
 #define OSD_PG_CREATING_PREFIX "osd_pg_creating"
 
+void LastEpochClean::Lec::report(ps_t ps, epoch_t last_epoch_clean)
+{
+  if (epoch_by_pg.size() <= ps) {
+    epoch_by_pg.resize(ps + 1, 0);
+  }
+  if (epoch_by_pg[ps] >= last_epoch_clean) {
+    // stale lec
+    return;
+  }
+  epoch_by_pg[ps] = last_epoch_clean;
+  if (last_epoch_clean < floor) {
+    floor = last_epoch_clean;
```

Comment:
It's a bit late, but this part here is a bug @tchaikov: there might be another PG whose lec is between floor and ps' last_epoch_clean, so updating the pool's floor requires doing a scan over the PGs. Right?